### PR TITLE
[ROCKETMQ-284] ExpressionMessageFilter will pass some message.

### PR DESCRIPTION
**Pull request by vsair**
https://issues.apache.org/jira/browse/ROCKETMQ-284

Solution:

Keep the tagscode transfer into MessageFilter is legal.

### CODE REVIEW

**Comment by vsair** [comment]
@vongosling  pls have a review.

File: broker/src/test/java/org/apache/rocketmq/broker/filter/MessageStoreWithFilterTest.java [hunk 1]
```diff
@@ -369,4 +367,36 @@ public void testGetMessage_withFilterBitMap() {
...
         }
     }
+
+    @Test
+    public void testGetMessage_withFilter_checkTagsCode() {
+        try {
+            putMsg(master, topicCount, msgPerTopic);
+            // sleep to wait for consume queue has been constructed.
+            Thread.sleep(200);
+        } catch (Exception e) {
```
Comment (vongosling) [inline broker/src/test/java/org/apache/rocketmq/broker/filter/MessageStoreWithFilterTest.java]: Could we using expected  for exception test ?

File: broker/src/test/java/org/apache/rocketmq/broker/filter/MessageStoreWithFilterTest.java [hunk 2]
```diff
@@ -369,4 +367,36 @@ public void testGetMessage_withFilterBitMap() {
             }
         }
     }
+
+    @Test
+    public void testGetMessage_withFilter_checkTagsCode() {
+        try {
+            putMsg(master, topicCount, msgPerTopic);
+            // sleep to wait for consume queue has been constructed.
```
Comment (vongosling) [inline broker/src/test/java/org/apache/rocketmq/broker/filter/MessageStoreWithFilterTest.java]: Must we use sleep to replace sleep ? Now ,we have polished our unit test within 3 minutes, thoughts @shroman

Comment (vsair) [reply broker/src/test/java/org/apache/rocketmq/broker/filter/MessageStoreWithFilterTest.java]: sleep to replace sleep ?

File: broker/src/test/java/org/apache/rocketmq/broker/filter/MessageStoreWithFilterTest.java [hunk 3]
```diff
@@ -79,24 +79,17 @@
         try {
             StoreHost = new InetSocketAddress(InetAddress.getLocalHost(), 8123);
         } catch (UnknownHostException e) {
-            e.printStackTrace();
         }
         try {
             BornHost = new InetSocketAddress(InetAddress.getByName("127.0.0.1"), 0);
         } catch (UnknownHostException e) {
```
Comment (vongosling) [inline broker/src/test/java/org/apache/rocketmq/broker/filter/MessageStoreWithFilterTest.java]: no need to delete  catch part ?

File: broker/src/test/java/org/apache/rocketmq/broker/filter/MessageStoreWithFilterTest.java [hunk 4]
```diff
@@ -79,24 +79,17 @@
...
     @Before
-    public void init() {
+    public void init() throws Exception {
         filterManager = ConsumerFilterManagerTest.gen(topicCount, msgPerTopic);
-        try {
-            master = gen(filterManager);
-        } catch (Exception e) {
-            e.printStackTrace();
-            assertThat(true).isFalse();
-        }
```
Comment (vongosling) [inline broker/src/test/java/org/apache/rocketmq/broker/filter/MessageStoreWithFilterTest.java]: Excellent !

File: broker/src/test/java/org/apache/rocketmq/broker/filter/MessageStoreWithFilterTest.java (hunk 3 above)
Comment (vsair) [reply broker/src/test/java/org/apache/rocketmq/broker/filter/MessageStoreWithFilterTest.java]: Here,  don't  care about this exception, so not hide it .

**Pull request closed without merging**